src/hints/execution.rs

```diff
@@ -737,18 +705,13 @@ pub fn tx_paymaster_data(
     _constants: &HashMap<String, Felt252>,
 ) -> Result<(), HintError> {
     let tx = exec_scopes.get::<InternalTransaction>(vars::scopes::TX)?;
-    // TODO: implement tx.version >= 3
-    assert!(tx.version.unwrap_or_default() < 3.into(), "tx.version >= 3 is not supported yet");
-
-    // let paymaster_data = if tx.version.unwrap_or_default() < 3.into() {
-    //     0.into()
-    // } else {
-    //     let paymaster_data = tx.paymaster_data.unwrap_or_default().iter().map(|felt|
-    // felt.into()).collect();     let paymaster_data_base = vm.add_memory_segment();
-    //     vm.load_data(paymaster_data_base, &paymaster_data)?;
-    //     paymaster_data_base
-    // };
-    let paymaster_data = Felt252::ZERO;
+
+    let paymaster_data = if tx.version.unwrap_or_default() < Felt252::THREE {
+        MaybeRelocatable::Int(Felt252::ZERO)
+    } else {
+        let paymaster_data_base = vm.add_memory_segment();
```

Comment:
This is just incorrect, you need to call `vm.gen_arg(&tx.paymaster_data)`, probably with some transformation of `tx.paymaster_data` to use the right types.